README.md

```diff
@@ -1,236 +1,261 @@
+
+## Use in DOFLinx
+
+To generate effects, DOFLinx uses `.MAME` files located in DOFLinx's MAME
+folder. DOFLinx can communicate with DOF2DMD, using DOFLinx `FF_DMD` command.
+The `FF_DMD` command can call any of the DOF2DMD APIs.
+
+### `DOFLinx.ini` file
+
+Here is a minimal DOFLinx.ini file which will work with `DOF2DMD`:
+
+```ini
+# location of your files and systems
+COLOUR_FILE=<DOFLinx path>\config\colours.ini
+DIRECTOUTPUTGLOBAL=<DOFLinx path>\config\GlobalConfig_b2sserver.xml
+PATH_MAME=<DOFLinx path>\MAME\
+MAME_FOLDER=<MAME executable path (note: it must be DOFLinx modified MAME version)>
+
+# When to activate, and more specifically what is the MAME process to kick things off
+PROCESSES=Mame64
+MAME_PROCESS=Mame64
+
+# DOF2DMD
+PATH_DOF2DMD=<location of DOF2DMD executable and settings.ini>
+```
+
+Note:
+
+- `PATH_DOF2DMD`: the location of DOF2DMD executable and settings.ini
+- `MAME_FOLDER`: MAME executable path which must be DOFLinx's [modified version of MAME](https://drive.google.com/drive/folders/1AjJ8EQo3AkmG2mw7w0fLzF9HcOjFoUZH)
+
+### Embedded commands
+
+DOFLinx will generate the following commands automatically:
+
+- When starting DOFLinx:
+  - `http://<host:port>/v1/version` - to check that DOF2DMD is up. DOFLinx will attempt to start it otherwise.
+  - `http://<host:port>/v1/display/picture?path=mame/DOFLinx` - to display the DOFLinx welcome picture
+- When starting a game:
+  - `http://<host:port>/v1/display/picture?path=mame/<rom-name>&duration=<duration>&animation=<animation>` - to display a PNG for the marquee
+- When playing a game:
+  - `http://<host:port>/v1/display/score?player=<active player>&score=<score>&cleanbg=<true|false>` - to display score of the given player
+  - `http://<host:port>/v1/display/scorev2?players=<number of players>&activeplayer=<active player>&score=<score>&cleanbg=<true|false>&credits=<credits>` - to display score of the given player inidicating the score board layout based on the number of players
+- When closing DOFLinx:
+  - `http://<host:port>/v1/display/score?player=1&score=0` - reset score to 0
+  - `http://<host:port>/v1/blank` - to clear the DMD (goes to black)
+  - `http://<host:port>/v1/exit` - to instruct DOF2DMD to exit cleanly
+
+### Syntax of `FF_DMD` DOFLinx command
+
+To add effects like showing animations or text during the game, you must insert
+the `FF_DMD` command in the `<rom>.MAME` file which corresponds to the game.
+
+```ascii
+FF_DMD,U,<DOF2DMD API CALL without host nor /v1/ prefix>
+```
+
+- `FF_DMD` is the command
+- `U` is for a user command (DOFLinx specific)
+- Then the URI to call DOF2DMD without host nor /v1/ prefix
+
+Examples :
+
+- Display the ingame bonus animation `artwork/ingame/bonus.gif` : `FF_DMD,U,display/picture?path=ingame/bonus&duration=0&animation=none`
+- Display a static picture `artwork/mame/pacman.png` : `FF_DMD,U,display/picture?path=mame/pacman&duration=-1`

```

Comment:
Same as above. Can we automatically set duration=-1 when MAME file says fixed=true?

Otherwise, it is fine to now use this format going forward. We don't change this README.md, just ensure backward compatibility in the code when receiving `fixed=true/false`.

Reply:
to be added in the code

Reply:
done